lisa/tools/modprobe.py

```diff
@@ -154,6 +162,58 @@ def load_by_file(
             result.assert_exit_code(0, f"failed to load module {file_name}")
         return result
 
+    def _debug_modprobe_failure(self, mod_name: str) -> None:
+        self._log.info(
+            f"Failed to remove module {mod_name}."
+            " Collecting additional debug information."
+        )
+        self._log_lsmod_status(mod_name)
+        self._tail_dmesg_log()
+        self._tail_journalctl_for_module(mod_name)
+
+    def _log_lsmod_status(self, mod_name: str) -> None:
+        lsmod_tool = self.node.tools[Lsmod]
+        module_exists = lsmod_tool.module_exists(
+            mod_name=mod_name,
+            force_run=True,
+            no_debug_log=True,
+        )
+        if not module_exists:
+            self._log.info(f"Module {mod_name} does not exist in lsmod output.")
+        else:
+            self._log.info(f"Module {mod_name} exists in lsmod output.")
+            used_by_modules = lsmod_tool.get_used_by_modules(mod_name)
+            if used_by_modules:
+                used_by_str = ", ".join(used_by_modules)
+                self._log.info(f"Module {mod_name} is used by: " f"{used_by_str}")
+            else:
+                self._log.info(f"Module {mod_name} is not used by any other modules.")
+
+    def _tail_dmesg_log(self) -> None:
+        dmesg_tool = self.node.tools[Dmesg]
+        dmesg_output = dmesg_tool.get_output(force_run=True, no_debug_log=True)
+        dmesg_lines = dmesg_output.splitlines()
+        tail_lines = 20
```

Comment:
how about pass this as parameter of one method of dmesg? then handle logic inside of dmesg tool?